test/controllers/import.test.js

```diff
@@ -192,6 +193,45 @@ describe('ImportController tests', () => {
       expect(response.headers.get('x-error')).to.equal('Invalid request: urls must be provided as a non-empty array');
     });
 
+    it('should fail when url is not part of the allowed domains', async () => {
+      baseContext.multipartFormData.urls = ['https://test.com/page1'];
+      const response = await importController.createImportJob(baseContext);
+
+      expect(response.status).to.equal(400);
+      expect(response.headers.get('x-error')).to.equal('Invalid request: URLs not allowed: https://test.com/page1');
+    });
+
+    it('should fail when there are no domains listed for the user scope imports.write', async () => {
+      baseContext.attributes.authInfo.profile.getScopes = () => [{ name: 'imports.write', domains: [] }];
+      const response = await importController.createImportJob(baseContext);
+
+      expect(response.status).to.equal(401);
+      expect(response.headers.get('x-error')).to.equal('Missing domain information');
+    });
+
+    it('should create an import job for the user scope imports.write', async () => {
+      baseContext.attributes.authInfo.profile.getScopes = () => [{ name: 'imports.write', domains: ['https://www.example.com'] }, { name: 'imports.read', domains: ['https://www.example.com'] }];
```

Comment:
> { name: 'imports.read', domains: ['https://www.example.com']

Is `domains` needed/supported on `imports.read`?

Reply:
Currently we do not need it. I added this unit test to test that the code picks up domains from only `WRITE` scope and not `READ`.